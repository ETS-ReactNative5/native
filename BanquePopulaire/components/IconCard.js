import React from 'react';
import { StyleSheet, View, Image, Text } from "react-native";

export default class IconCard extends React.Component {
    render() {
        return (
            <View style={styles.container}>
                <Image style={styles.icon} source={this.props.source}/>
                <View style={styles.title}>
                    <Text style={{color: 'white', fontSize: 26, textAlign: 'center'}}>{this.props.content}</Text>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        alignContent: 'center',
        justifyContent: 'center',
        backgroundColor: "#094fa3ff",
        width: 300,
        height: 300,
        margin: 20,
        padding: 0,
    },

    icon: {
        width: 180,
        height: 180,
        alignSelf: 'center',
    },

    title: {
        backgroundColor: '#0000002b',
        marginLeft: 16,
        marginRight: 16,
        marginTop: 16,
        padding: 25,
    }
})