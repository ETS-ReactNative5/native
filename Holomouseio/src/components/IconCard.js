import React from 'react';
import { StyleSheet, View, Image, Text } from "react-native";

export default class IconCard extends React.Component {
    render() {
        return (
            <View style={{...styles.container, ...this.props.style}}>
                <Image style={{alignSelf: 'center', width: this.props.style.width * 0.6 ||Â 0, height: this.props.style.height * 0.6 || 0, marginTop: 16}} source={this.props.source}/>
                <View style={styles.title}>
                    <Text style={{color: 'white', fontSize: this.props.style.titleSize || 26, textAlign: this.props.style.titleAlign || 'center'}}>{this.props.content}</Text>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        alignContent: 'center',
        justifyContent: 'center',
        margin: 20,
        padding: 8,
        borderRadius: 8
    },

    title: {
        marginLeft: 16,
        marginRight: 16,
        marginTop: 10,
    }
})