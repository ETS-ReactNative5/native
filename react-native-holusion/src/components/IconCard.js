import React from 'react';
import { StyleSheet, View, Image, Text } from "react-native";

export default class IconCard extends React.Component {
  render() {
    let imageStyle = {};
    let textStyle = {};
    let style = {...styles.container, ...this.props.style} || styles.container;

    if(this.props.source) {
      imageStyle = {alignSelf: 'center', width: style.width * 0.6 || 0, height: style.height * 0.6 || 0, marginTop: 16};
    } else {
      imageStyle = {alignSelf: 'center', width: 0, height: 0, marginTop: 16};
    }

    if(this.props.style) {
      textStyle = {color: 'white', fontSize: style.titleSize || 26, textAlign: style.titleAlign || 'center'};
    } else {
      textStyle = {color: 'white', fontSize: 26, textAlign: 'center'};
    }

    return (
        <View style={{...styles.container, ...this.props.style}}>
            <Image style={imageStyle} source={this.props.source}/>
            <View style={styles.title}>
                <Text style={textStyle}>{this.props.content}</Text>
            </View>
        </View>
    );
  }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        alignContent: 'center',
        justifyContent: 'center',
        margin: 20,
        padding: 8,
        borderRadius: 8,
        width: 200,
        height: 200
    },

    title: {
        marginLeft: 16,
        marginRight: 16,
        marginTop: 10,
    }
})
