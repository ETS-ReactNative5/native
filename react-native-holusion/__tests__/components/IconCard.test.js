import React from 'react'
import IconCard from '../../src/components/IconCard'

import renderer from 'react-test-renderer'

test('renders IconCard correctly', () => {
  let tree = renderer.create(<IconCard />).toJSON();
  expect(tree).toMatchSnapshot();

  tree = renderer.create(<IconCard content="title" />).toJSON();
  expect(tree).toMatchSnapshot();

  tree = renderer.create(<IconCard content="title" source={require("./test.jpg")} />).toJSON();
  expect(tree).toMatchSnapshot();
})
