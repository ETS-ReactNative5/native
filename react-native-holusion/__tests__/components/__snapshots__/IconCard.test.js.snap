// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IconCard correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "column",
      "height": 200,
      "justifyContent": "center",
      "margin": 20,
      "padding": 8,
      "width": 200,
    }
  }
>
  <Image
    style={
      Object {
        "alignSelf": "center",
        "height": 0,
        "marginTop": 16,
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders IconCard correctly 2`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "column",
      "height": 200,
      "justifyContent": "center",
      "margin": 20,
      "padding": 8,
      "width": 200,
    }
  }
>
  <Image
    style={
      Object {
        "alignSelf": "center",
        "height": 0,
        "marginTop": 16,
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "textAlign": "center",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`renders IconCard correctly 3`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "column",
      "height": 200,
      "justifyContent": "center",
      "margin": 20,
      "padding": 8,
      "width": 200,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../__tests__/components/test.jpg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 120,
        "marginTop": 16,
        "width": 120,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "textAlign": "center",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
